src/GovUk.Education.ExploreEducationStatistics.Data.Services/ObservationService.cs

```diff
@@ -109,51 +109,52 @@ public IEnumerable<Observation> FindObservations(ObservationQueryContext query)
                     planningAreaListParam,
                     filterItemListParam);
 
-            _logger.LogTrace("Executed inner stored procedure in {Time} ms", stopwatch.Elapsed.TotalMilliseconds);
-            stopwatch.Restart();
+            _logger.LogDebug($"Executed FilteredObservations stored procedure in {phasesStopwatch.Elapsed.Milliseconds} ms");
+            phasesStopwatch.Restart();
 
             var ids = inner.Select(obs => obs.Id).ToArray();
 
-            _logger.LogTrace("Fetched Observation id's from inner result in {Time} ms", stopwatch.Elapsed.TotalMilliseconds);
-            stopwatch.Restart();
-
-            var batches = ids.Batch(10000);
+            _logger.LogDebug($"Fetched {ids.Length} Observation ids from inner result in {phasesStopwatch.Elapsed.Milliseconds} ms");
+            phasesStopwatch.Restart();
+            
+            var observations = _context
+                .Observation
+                .AsNoTracking()
+                .Include(o => o.FilterItems)
+                .Where(o => ids.Contains(o.Id))
+                .ToList();
 
-            var result = new List<Observation>();
+            _logger.LogDebug($"Fetched {ids.Length} Observations from their ids in {phasesStopwatch.Elapsed.Milliseconds} ms");
+            phasesStopwatch.Restart();
 
-            foreach (var batch in batches)
```

Comment:
Pretty sure during the development of this that it was erroring on the database with a large number of results, hence the batching of `ids` into 10,000 at a time. Have you tried this with a large subject file like `absence_by_characteristic.csv` with 200,000 rows, and run a query that would return all or most of those rows like selecting a large number of the characteristics and all the school types?